Symbols Approach for Addressables

Selva1910 opened the pull request:
Modification on - Uncomment for Addressables as a Setting in Build Settings Editor Window

## Description
Used Script Symbols, So that it can be toggle on and off with out much effort. It will get all the available targets present and add a symbol when you press apply.
 
## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

Selva1910 committed:
Symbols Approach for Addressables

Modifed UnComment for Addressables code into a Settings in Build Settings Editor Window

Selva1910 commented:
I hope this is kind of easy to preview or split the changes that i am making ?

Selva1910 committed:
Removed Unwanted .vscode folder data

timbotimbo commented:
Thanks, this is much easier to understand,

You won't have to make a separate PR for each and every small feature, just make sure you can easily see what actually changed.

I did actually sync the Unity scripts in master with the 6000 branch after your PR. So a PR to master is fine too.  
Sorry for not letting you know.
##

I tried this in my own project that already uses addressables, but where I've always manually built them so far.
With some small modifications I was able to export and the addressable assets were properly included in my build.

To get it to work I needed to:

Remove these comments:
```c#
#if USING_ADDRESSABLES
// uncomment for addressables
//using UnityEditor.AddressableAssets;
//using UnityEditor.AddressableAssets.Settings;
#endif
```

And I needed to add an `Assembly Definition Reference` to `FlutterUnityInteration.Editor.asmdef`.
```asmdef
"references": [
        "Unity.Addressables.Editor"
    ],
```
Having this reference (by name, not GUID) in a project without the Addressables package installed makes it show up greyed out. It doesn't cause any errors and installing the Addressables package enables it again.
So that should be safe to include by default.

![asmdef](https://github.com/user-attachments/assets/7014ff9d-ca72-4b02-ac17-06d30996621c)


##
One small issue seems to be that the state of the checkbox isn't always correct.
Sometimes `USING_ADDRESSABLES` is enabled but the checkbox shows up as unchecked when opening the settings menu.
Simply restarting Unity can cause this. Removing the define symbol in the player settings makes this happen too.

Maybe this state can be directly loaded from the existence of `USING_ADDRESSABLES`?

Selva1910 commented:
Cool I will note these things down. And I will update the full changes this was raised most probably just to verify everything is fine for reviewing.

Selva1910 committed:
Required Bug Fixes

- Prefs not Initializing so that the toggle state was not synced
- Uncommented Addressables Headers
- Added Name References for Addressables to Editor AsmDef

Selva1910 committed:
Folder Organization for Packages Conversion

timbotimbo commented:
Sorry for the long delay. I'll make sure to get this sorted this week.

timbotimbo commented:
Just did some tests to confirm:
- It still functions installed as a classic unitypackage.
- Can install using package manager local path.
- Can be installed using package manager git url. 
  Using the following url for now.
```
https://github.com/Selva1910/flutter-unity-view-widget.git?path=example/unity/DemoApp/Assets/FlutterUnityIntegration#experimental/unity_6000
```



- Builds from the command line still work with the new assembly definitions and a package manager install.
```
<unity path> -projectPath <project path> -batchmode -buildTarget Android -executeMethod FlutterUnityIntegration.Editor.Build.DoBuildAndroidLibraryRelease -quit
```
- All script compile in Unity 6000 and 2022.

##
I spotted a small issue when importing in older legacy Unity versions, but i'll look into that myself.

Selva1910 commented:
Since you have merged it now, it is working on this git url from now onwards.
```
https://github.com/juicycleff/flutter-unity-view-widget.git?path=example/unity/DemoApp/Assets/FlutterUnityIntegration#experimental/unity_6000
```
